### ISSUE TITLE
docker image or the pip env list?

### ISSUE THREAD

**Issue by yangshunDragon** [wilkinghoff/sub-cluster-AdaCos#2]
Hello,  @wilkinghoff 
Thanks very much for opening source your code!

I reimplement the env and tried to run with my own audio dataset.
It seems there exist some bug for keras or  tf.keras?  https://github.com/wilkinghoff/sub-cluster-AdaCos/blob/52d6758d68df7744a49e76bebeae5c9d2b221ba3/main.py#L387  
Maybe I used different version for some lib in my env.
Could you please share your docker image or the pip env list? Then we could tell what is different.

**Comment by yangshunDragon** [comment]
My pip env list is as below:
Package                 Version
----------------------- -------------------
absl-py                 0.15.0
appdirs                 1.4.4
astor                   0.8.1
astunparse              1.6.3
audioread               2.1.9
cached-property         1.5.2
cachetools              4.2.4
certifi                 2021.10.8
cffi                    1.15.0
charset-normalizer      2.0.9
clang                   5.0
cloudpickle             2.0.0
dataclasses             0.8
decorator               5.1.0
flatbuffers             1.12
gast                    0.2.2
google-auth             1.35.0
google-auth-oauthlib    0.4.6
google-pasta            0.2.0
grpcio                  1.43.0
h5py                    3.1.0
idna                    3.3
importlib-metadata      4.8.3
joblib                  1.1.0
Keras                   2.3.1
Keras-Applications      1.0.8
Keras-Preprocessing     1.1.2
librosa                 0.8.1
llvmlite                0.36.0
Markdown                3.3.6
normalSpeed             0.0.1
numba                   0.53.1
numpy                   1.19.5
oauthlib                3.1.1
opt-einsum              3.3.0
packaging               21.3
pandas                  1.1.5
pip                     21.3.1
pooch                   1.5.2
protobuf                3.19.1
pyasn1                  0.4.8
pyasn1-modules          0.2.8
pycparser               2.21
pyparsing               3.0.6
python-dateutil         2.8.2
pytz                    2021.3
PyYAML                  6.0
requests                2.26.0
requests-oauthlib       1.3.0
resampy                 0.2.2
rsa                     4.8
scikit-learn            0.24.2
scipy                   1.4.1
setuptools              49.6.0.post20210108
six                     1.15.0
SoundFile               0.10.3.post1
tensorboard             2.1.1
tensorboard-data-server 0.6.1
tensorboard-plugin-wit  1.8.0
tensorflow              2.1.0
tensorflow-estimator    2.1.0
tensorflow-gpu          2.1.0
tensorflow-probability  0.9.0
termcolor               1.1.0
threadpoolctl           3.0.0
tqdm                    4.62.3
typing-extensions       3.7.4.3
urllib3                 1.26.7
Werkzeug                2.0.2
wheel                   0.37.1
wrapt                   1.12.1
zipp                    3.6.0

**Comment by wilkinghoff** [comment]
Here is mine:
Package                Version
---------------------- -------------------
absl-py                0.9.0
appdirs                1.4.4
astor                  0.6.2
astunparse             1.6.3
audioread              2.1.6
aupyom                 0.1.0
backcall               0.1.0
bleach                 1.5.0
cachetools             4.0.0
certifi                2018.10.15
cffi                   1.11.5
chardet                3.0.4
Click                  7.0
cloudpickle            1.4.1
colorama               0.4.1
cycler                 0.10.0
Cython                 0.29.21
dcase-util             0.2.10
decorator              4.3.0
deprecation            2.0.7
dm-tree                0.1.5
docopt                 0.6.2
flatbuffers            1.12
future                 0.17.1
gast                   0.3.3
google-auth            1.11.3
google-auth-oauthlib   0.4.1
google-pasta           0.2.0
grpcio                 1.27.2
gym                    0.10.9
h5py                   2.10.0
hdf5storage            0.1.15
html5lib               0.9999999
idna                   2.7
imageio                2.8.0
imageio-ffmpeg         0.4.2
imbalanced-learn       0.8.0
imblearn               0.0
ipython                7.5.0
ipython-genutils       0.2.0
jedi                   0.13.3
jiwer                  2.1.0
joblib                 0.17.0
kapre                  0.1.4
Keras                  2.2.5
Keras-Applications     1.0.8
Keras-Preprocessing    1.1.2
kiwisolver             1.0.1
librosa                0.8.0
llvmlite               0.34.0
Markdown               2.6.11
matplotlib             2.2.2
ml-metrics             0.1.4
mock                   2.0.0
moviepy                1.0.3
networkx               2.4
nose                   1.3.7
np-utils               0.5.12.1
num2words              0.5.10
numba                  0.51.2
numpy                  1.18.2
oauthlib               3.1.0
opencv-contrib-python  3.4.3.18
openl3                 0.3.1
opt-einsum             3.2.0
packaging              20.3
pandas                 0.23.0
parso                  0.4.0
pbr                    4.0.4
pickleshare            0.7.5
Pillow                 5.3.0
pip                    20.2.4
pooch                  1.2.0
proglog                0.1.9
prompt-toolkit         2.0.9
protobuf               3.11.3
pyaml                  19.12.0
pyasn1                 0.4.8
pyasn1-modules         0.2.8
pycparser              2.19
pydot-ng               2.0.0
pyglet                 1.3.2
Pygments               2.4.2
PyOpenGL               3.1.0
pyparsing              2.2.0
PySoundFile            0.9.0.post1
python-crfsuite        0.9.7
python-dateutil        2.7.3
python-Levenshtein     0.12.0
python-magic           0.4.15
python-speech-features 0.6
pytz                   2018.4
PyWavelets             1.1.1
PyYAML                 5.1
requests               2.23.0
requests-oauthlib      1.3.0
resampy                0.2.2
rsa                    4.0
scikit-image           0.14.5
scikit-learn           0.24.2
scipy                  1.4.1
seaborn                0.11.0
sed-eval               0.2.1
setuptools             46.0.0
SIDEKIT                1.2.3
six                    1.14.0
sklearn                0.0
sklearn-crfsuite       0.3.6
snips-nlu              0.20.2
snips-nlu-en           0.2.3
snips-nlu-parsers      0.4.3
snips-nlu-utils        0.9.1
sounddevice            0.3.12
SoundFile              0.10.2
sphfile                1.0.2
swalign                0.3.6
tabulate               0.8.7
tb-nightly             2.4.0a20200818
tensorboard            2.3.0
tensorboard-plugin-wit 1.6.0.post3
tensorboard-pytorch    0.7.1
tensorboardX           1.4
tensorflow             2.3.0
tensorflow-addons      0.11.2
tensorflow-estimator   2.3.0
tensorflow-probability 0.11.1
tensorflow-tensorboard 1.5.1
termcolor              1.1.0
tf-estimator-nightly   2.3.0.dev2020061601
Theano                 1.0.2
threadpoolctl          2.1.0
torch                  0.4.1
torchvision            0.2.1
tqdm                   4.32.1
traitlets              4.3.2
tslearn                0.5.0.5
typeguard              2.10.0
urllib3                1.24.1
validators             0.13.0
wcwidth                0.1.7
Werkzeug               0.14.1
wheel                  0.31.1
wrapt                  1.12.1
